### DISCUSSION TITLE
Language options to import from nominatim to photon

### DISCUSSION THREAD

**Discussion by xinyuluo** [komoot/photon#654] [Q&A] [answered]
Hi, I hope to import all available languages from nominatim database, but not sure what's available. When I run:

`java -jar photon-*.jar -nominatim-import -host localhost -port 5432 -database nominatim -user nominatim -password mysecretpassword`

it seems like only English, German, France and Italian are imported. If these are the default options, how can i get a list of languages / import full list of languages to photon?

Thank you 🙏

**Comment by mtmail** [comment, marked as answer]
You'd have to specify each language code in a comma separated list, e.g. `-languages es,fr,de`. In theory that's every 2 and 3 letter https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes 

For example on https://www.openstreetmap.org/relation/148838 you can see tags `name:rw`, `name:syc` and the import will then use those values.

**Reply by lonvia** [reply]
To add to that: https://taginfo.openstreetmap.org/search?q=name%3A is a good source to get an idea which languages are used in OSM.

**Reply by xinyuluo** [reply]
Thank you @mtmail @lonvia . Just a follow-up question, is there a way to check which languages have been imported to Photon? Thanks

**Reply by xinyuluo** [reply]
Sorry, could I please get an update on the follow-up question (in case it was skipped accidentally)? Thank you.

**Reply by lonvia** [reply]
No, there is no easy way to check.

**Comment by lonvia** [comment]
Please be aware that Photon does not work well against many languages at them moment. See the discussion in #615 for background.

**Reply by xinyuluo** [reply]
@lonvia thanks for the information and link to the discussion. To clarify,
1. Based on the discussion thread, it seems like photon builds a query that tries to search across all supported languages for an incoming request. Does this mean, the more languages i import from Nominatim, the more language photon supports, and as a result it might become slower?
2. I didn't closely read code between Photon and Nominatim. Does Nominatim work better with different languages, or have similar multi-language support as Photon at the moment?
Thank you.

**Reply by lonvia** [reply]
1. Yes.
2. Nominatim's search algorithm is completely different with respect to languages. It has no restriction regarding the number of different names.
